### ISSUE TITLE
Oops! We encountered an error. Please try again.

### ISSUE THREAD

**Issue by smentfamily** [tonesto7/nest-manager#488]
I receive the error above when trying to add Nest Manager to My Apps on my android phone. The error occurs after successfully entering in my Nest email address and password and selecting next. The interstitial page is "Works with Nest" and describes "Home would like to do the following." Scrolling down the page and entering "Agree" produces the above error.

This it the GitHub log...

bfa06010-ac91-499e-bfbf-083e92fd540b  12:13:18 AM: warn || Nest Manager | Missing: structData null
bfa06010-ac91-499e-bfbf-083e92fd540b  12:13:13 AM: info ||| Nest Manager | AuthToken not found: Directing to Login Page
bfa06010-ac91-499e-bfbf-083e92fd540b  12:13:13 AM: info ||| Nest Manager | appData.json File HAS Changed
2b39e8e2-881e-460a-99de-42391568303d  12:13:02 AM: warn DEPRECATION: You no longer need to call 'unsubscribe' within the 'uninstalled' method.
2b39e8e2-881e-460a-99de-42391568303d  12:12:53 AM: info ||| Nest Manager | AuthToken not found: Directing to Login Page
38aaa56b-b73f-4b94-8617-0a2cc90acb35  12:12:09 AM: debug getChildDevices(false), children=0
38aaa56b-b73f-4b94-8617-0a2cc90acb35  12:12:09 AM: debug detecting status for f4ebf2c7-0ccc-4c54-9611-39fc0ce80d55 : false
38aaa56b-b73f-4b94-8617-0a2cc90acb35  12:12:09 AM: trace in ssdpDiscover
2b39e8e2-881e-460a-99de-42391568303d  12:11:34 AM: warn || Nest Manager | Missing: structData null
2b39e8e2-881e-460a-99de-42391568303d  12:11:29 AM: info ||| Nest Manager | AuthToken not found: Directing to Login Page
a530851c-9e86-484c-92c3-dfbe1e4bf3bf  12:11:13 AM: debug getChildDevices(true), children=0
2b39e8e2-881e-460a-99de-42391568303d  12:10:07 AM: warn || Nest Manager | Missing: structData null
2b39e8e2-881e-460a-99de-42391568303d  12:10:05 AM: info ||| Nest Manager | AuthToken not found: Directing to Login Page
 12:09:50 AM: info Waiting on events...
 12:09:50 AM: info For past logs for individual things go to the My Devices section, find the device and click on the Events link on the device information page.
 12:09:50 AM: info This console provides live logging of your SmartThings.

**Comment by shineet** [comment]
I get the same error. Let me know if you are able to resolve it

**Comment by bjhale** [comment]
I am getting the same error. I am even using my own nest developer account.

**Comment by shineet** [comment]
Looks like 3 of us are facing the same issue. Can some one please help us

**Comment by tonesto7** [comment]
Sounds like your not pressing update after you copy in the id and secret. 
Either way i've added 50 more tokens so give it a try again

**Comment by smentfamily** [comment]
The tokens must have been the solution. Thank you!
________________________________

**Comment by shineet** [comment]
Thanks a ton. It worked.

**Comment by Mjclar4** [comment]
I'm getting the error message still.

**Comment by oscarwu1991** [comment]
me too. Please release more tokens :)

**Comment by oscarwu1991** [comment]
@tonesto7 please release more tokens

**Comment by tonesto7** [comment]
I don't have anymore... You will need to create your own.  Instructions can be found in the wiki page

**Comment by l1brass** [comment]
Tired using documentation to create nest developer acct and my own product.  Then stored the clientid and secret in smartapp, updated, saved, and published.  Getting the oops error before I get to the agree screen now.  Checked the instructions MANY times but can't figure out where I went wrong.  No help from the logs just states that there is no AuthToken.   Must be missing something here, any suggestions?

**Comment by callrw** [comment]
I am getting a Error and cannot for the life of me figure out why : redirect_uri mismatch -- the rediredt URI in the request https:graph.api.smarthings.com/oauth/callback does not match the ones authorized for the OAUTH client. Visit the developer console to update the authorized redirect URI

**Comment by blazchasr** [comment]
I've run into the same error as 1brass doing the same process.

**Comment by marksthomas** [comment]
Yeah, same issue here. This is so annoying. I've created my own app. Got the codes. Pasted them as per the google doc, and still OOOPS! etc etc. Someone must know what is going on.,

This is in the logs..

 ||| Nest Manager | AuthToken not found: Directing to Login Page

The bloody thing is there. So frustrating. 

Any ideas?

**Comment by tonesto7** [comment]
Guys come on... I can only do so much... Do a search in the community or here.   It's be said that you need to create your own tokens now as i'm out and Nest won't give me any more... We are at like 7000 tokens. 
I haven't had time or ambition lately to remove our token completely.

**Comment by marksthomas** [comment]
Yeah, I've already done the create token (as I said) and same issue.

I've done so much searching that frankly I can't be arsed any more.

Sorry to put you out.

**Comment by tonesto7** [comment]
well obviously you didn't insert the values correctly or it would work like everyone else's.
Did you verify the redirect URL in Nest Developer portal is correct?  Did you make sure there wasn't any white spaces in the id and secret you copied in the IDE.
How am i putting you out?  I frickin' automated everything I could.

**Comment by marksthomas** [comment]
Sorry to put **_YOU_** out. I was apologising for being annoying.

Thank you. I'll check the things you said.

**Comment by dlloyd242** [comment]
Was having the same issue, followed the instructions to create the Nest token, worked perfectly.  Thank you!

**Comment by marksthomas** [comment]
Wish mine did.

What was your redirect URL? Mine gives error 500 when testing...

**Comment by tonesto7** [comment]
@marksthomas PM in the ST community forum @tonesto7.  Please send me a screen shot of your nest developer page and another of the NST Settings Page in IDE (Same page you would enable Oauth).

**Comment by marksthomas** [comment]
Here you go. Like I said, sorry for being a pain, but I can't work it out and it makes my head hurt ;) It looked really straight forward, so I must be  missing something silly.

![st](https://user-images.githubusercontent.com/32539594/31279316-8f59a6b4-aaa0-11e7-98a1-21397cd328a1.png)
![nest](https://user-images.githubusercontent.com/32539594/31279317-8f60a414-aaa0-11e7-88e0-daf061247291.png)

**Comment by tonesto7** [comment]
the documentation never showed you to modify any of the oauth items.  So you didn't follow the 
directions.  
You didn't need to play with any of the oauth.  You were just supposed to copy the values into the value boxes in the settings for the clientId and clientSecret, you overwrote the keys and left the values blank.
Delete the enter app from the IDE and start over except this time please follow the document step by step and make sure the items look like those in the pictures.

**Comment by marksthomas** [comment]
BOOM!

See. Told you it was something silly. The documentation mentioned"-

Product ID will go in clientId
Product Secret will go in clientSecret

I took it literally and put in those fields (oops) - the redirect URL was me playing - trying to get it to work.

Now works. Donation inbound.

Thanks for that and appreciate the help.

**Comment by tonesto7** [comment]
Glad you got it working.  Any suggestions on making the documentation more error proof?

**Comment by marksthomas** [comment]
To be honest, it was plain sailing - really easy to follow - I just tripped up on putting the keys in the wrong field - perhaps a big arrow pointing out the /correct/ place for other duffuses like me would help.

I'm off to investigate it now! :)

**Comment by rsanchez007** [comment]
I still get an error after following all the comments.  Oops! We encountered an error. Please try again.

can anyone help please.  I can't login into the nest from the smarthings app.

**Comment by kennygray68** [comment]
Getting the oops error after entering password. Can you try to add tokens again?

**Comment by tonesto7** [comment]
#442 #
